src/ui/wxWidgets/TreeCtrl.cpp

```diff
@@ -993,6 +1028,20 @@ void TreeCtrl::RestoreGroupDisplayState()
   }
 }
 
+bool TreeCtrl::Show(bool show)
+{
+  /*
+    At least one tree element is needed to make the context menu work,
+    otherwise it won't show. TreeCtrl::AddGroup checks if a root item
+    is needed or an existing one is available.
+  */
+  if (IsEmpty()) {
+    AddRoot(wxT("root"));
```

Comment:
The `Show` method looks like a simple predicate, so it's a bit counter-intuitive that it may modify the database. Ideally any modifying operation should ensure it preserves the root so we won't have to do it here.